### ISSUE TITLE
change to agpl

### ISSUE THREAD

**Issue by cauerego** [lbryio/lbry-sdk#3473]
i got amazed at how searching for "agpl" brings zero issues!

as you know, we should always carefully choose the license before creating any software.

lbry and odyssey already got very big with MIT, and you probably haven't faced many issues yet...

and we have plenty of instances of free software using several different kinds of license...

but, imho, if we want to get real and serious about freedom, we need to go with agpl today. nothing less.

think about it. why no big corp ever picks agpl? what really lies on the bottom of the fuss against it?

i have (for now) zero affiliation with gnu or fsf... but i will get in because i truly believe in this. and i will get out once something better comes up, if ever. history, so far, showed zero better promises.

heads up, and..

cheers! 😁

**Comment by vertbyqb** [comment]
For this project I don't think the AGPL would be the right choice. The AGPL is meant for server software, so LBRY Desktop (has a web UI that Odysee and Madiator.com use) and commentron (the comment server) might be better candidate for the AGPL. The plain GPL is designed for non-web software. Though I suppose the only difference between the AGPL and the GPL is one clause, so maybe it's better to be safe than sorry?

Of course, even if all the licenses are changed people could still use the old version with the MIT license, but it would stop new proprietary software from being created, they would have to chose between a old version or giving the users freedoms.

To change the license all contributors would have to agree. I don't know if contributors need to assign copyright to LBRY Inc, but if not it could be a struggle getting everyone to agree to the license change.

**Comment by cauerego** [comment]
yes, a lot of contributors make it very difficult to change the license. and we have plenty of misinformation around licenses.

i do not know for sure, but in my understanding agpl indeed adds that one clause from gpl to protect the user freedom to access code from services as well, not just the app itself. so it triggers devs who want to use 3rd party APIs, for instance. i think with agpl you can't use an API that doesn't allow to see their code (which is usually the primary reason to build an API)...

but i also guess that lbry doesn't use any kind of such "API", so it should be fine.
